app/models/custom_attributes/custom_attribute_value.rb

```diff
@@ -20,6 +20,24 @@ def value_or_display_value
       def value_present?
         date_time_value.present? || string_value.present? || double_value.present? || integer_value.present? || !boolean_value.nil? || custom_attribute_option_values.any?
       end
+
+      def is_date?(date = date_time_value)
+        !!date.to_date rescue false
+      end
+
+      def date_for_next_alert
+        if !is_date?
+          nil
```

Comment:
in a date type alert can there be a string type value? ... if this a safety check then it should be a validation on the value model